packages/apiserver/src/resources/item/item.resolver.ts

```diff
@@ -19,8 +23,15 @@ export class ItemResolver {
   }
 
   @Mutation(() => Status)
-  async claimBook(@Args('input') claimBookInput: BookInput) {
-    return this.itemService.claim(claimBookInput);
+  async claimBook(
+    @Args('input') claimBookInput: BookInput
+  ): Promise<Status | ItemError> {
+    try {
+      return await this.itemService.claim(claimBookInput);
+    } catch (e) {
+      console.log(e);
+      return new ItemError(e.message);
```

Comment:
why we need to return an error here and below?

Reply:
It's defined in  graphql schema like this
```
mutation RejectItem($identifier: String!, $person_id: Int!) {
  rejectItem(input: { identifier: $identifier, person_id: $person_id }) {
    ... on Status {
      created_at
      status
    }
    ... on Error {
      message
    }
  }
}
```